lib/elixir/lib/version.ex

```diff
@@ -192,15 +212,16 @@ defmodule Version do
     match?(version, parse_requirement!(requirement), opts)
   end
 
-  def match?(version, %Requirement{matchspec: spec, compiled: false}, opts) do
-    allow_pre = Keyword.get(opts, :allow_pre, true)
-    {:ok, result} = :ets.test_ms(to_matchable(version, allow_pre), spec)
-    result != false
-  end
-
-  def match?(version, %Requirement{matchspec: spec, compiled: true}, opts) do
+  def match?(version, requirement, opts) do
+    spec = Requirement.matchspec(spec)
     allow_pre = Keyword.get(opts, :allow_pre, true)
-    :ets.match_spec_run([to_matchable(version, allow_pre)], spec) != []
+    if Requirement.compiled?(requirement) do
+      matches = :ets.match_spec_run([to_matchable(version, allow_pre)], spec)
+      matches != []
```

Comment:
Consideration: this is this way because it was based on the original implementation. Particularly I'd prefer to use `Enum.any?(matches)` because it sounds nicer :>